libethcore/SealEngine.h

```diff
@@ -140,6 +140,11 @@ class NoProof: public eth::SealEngineBase
 public:
 	std::string name() const override { return "NoProof"; }
 	static void init();
+    void populateFromParent(BlockHeader& _bi, BlockHeader const& _parent) const override
+    {
+        SealEngineFace::populateFromParent(_bi, _parent);
+        _bi.setDifficulty(chainParams().minimumDifficulty);
```

Comment:
I see that in both `c_genesisInfoFrontierNoProofTest` and `c_genesisInfoMainNetworkNoProofTest` genesis block difficulty is higher than minimumDifficulty.

Should we maybe for consistency change the difficulty of those genesis blocks to minimum? 
If we do that, then the easier fix here could be change the following line to `m_difficulty = _parent.m_difficulty;` https://github.com/ethereum/cpp-ethereum/blob/aa73807c9a6f79114d36ce738658dcba0bc7fbb3/libethcore/BlockHeader.cpp#L208

Reply:
This sounds reasonable. Let me run the tests with your change.

Reply:
```m_difficulty = _parent.m_difficulty``` does not work

Reply:
I think difficulty related tests should use Ethash. otherwise it is unclear what should be the return value of difficulty instruction. here it is set to c_minDifficulty. but I guess such tests should say to other clients that it should be executed with normal mining algo.

Reply:
Why didn't it work?